src/libcore/mem.rs

```diff
@@ -548,8 +548,19 @@ pub unsafe fn uninitialized<T>() -> T {
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub fn swap<T>(x: &mut T, y: &mut T) {
-    unsafe {
-        ptr::swap_nonoverlapping(x, y, 1);
+    if size_of::<T>() < 32 {
```

Comment:
Is it possible to make it so this function and `swap_nonoverlapping_bytes` are always in sync? e.g., If the latter is updated, can we also guarantee that the former also must be updated?